modules/criteoBidAdapter.js

```diff
@@ -33,6 +32,233 @@ const BUNDLE_COOKIE_NAME = 'cto_bundle';
+
+    // We remove the native asset requirements if we used the bypass to generate the imp
+    const hasAssetRequirements = requestNative.assets &&
+      (requestNative.assets.length !== 1 || Object.keys(requestNative.assets[0]).length);
+    if (!hasAssetRequirements) {
+      delete requestNative.assets;
+    }
+
+    deepSetValue(imp, 'native.request_native', requestNative);
+    delete imp.native.request;
+  }
+
+  return imp;
+}
+
+/**
+ * Builds a request object for the ORTB 2.5 request.
+ *
+ * @param {function} buildRequest - The function for building a request object.
+ * @param {Array} imps - An array of ORTB 2.5 impression objects.
+ * @param {Object} bidderRequest - The bidder request object.
+ * @param {Object} context - The context object.
+ * @returns {Object} The ORTB 2.5 request object.
+ */
+function request(buildRequest, imps, bidderRequest, context) {
+  let request = buildRequest(imps, bidderRequest, context);
+
+  if (typeof request.app !== 'undefined') {
+    // app
+    if (typeof context.publisherId !== 'undefined') {
+      deepSetValue(request, 'app.publisher.id', context.publisherId);
+    }
+  } else {
+    // site
+    if (typeof context.url !== 'undefined') {
+      deepSetValue(request, 'site.page', context.url);
+    }
+    if (typeof context.publisherId !== 'undefined') {
+      deepSetValue(request, 'site.publisher.id', context.publisherId);
+    }
+  }
+
+  if (bidderRequest.eids) {
+    deepSetValue(request, 'user.ext.eids', [...bidderRequest.eids]);
+  }
+
+  if (typeof deepAccess(request, 'device.sua') !== 'undefined') {
+    deepSetValue(request, 'device.ext.sua', request.device.sua);
+    delete request.device.sua; // moved 2.6 field to ext
+  }
+  if (typeof deepAccess(request, 'regs.gpp') !== 'undefined') {
+    deepSetValue(request, 'regs.ext.gpp', request.regs.gpp);
+    delete request.regs.gpp // moved 2.6 field to ext
+  }
+  if (typeof deepAccess(request, 'regs.gpp_sid') !== 'undefined') {
+    deepSetValue(request, 'regs.ext.gpp_sid', request.regs.gpp_sid);
+    delete request.regs.gpp_sid // moved 2.6 field to ext
+  }
+
+  if (bidderRequest) {
+    if (typeof bidderRequest.coppa !== 'undefined') {
+      deepSetValue(request, 'regs.coppa', bidderRequest.coppa === true ? 1 : (bidderRequest.coppa === false ? 0 : undefined));
+    }
+
+    if (bidderRequest.uspConsent) {
+      deepSetValue(request, 'regs.ext.us_privacy', bidderRequest.uspConsent);
+    }
+
+    if (bidderRequest.gdprConsent) {
+      deepSetValue(request, 'regs.ext.gdprversion', bidderRequest.gdprConsent.apiVersion);
+      if (typeof bidderRequest.gdprConsent.gdprApplies !== 'undefined') {
+        deepSetValue(request, 'regs.ext.gdpr', !!(bidderRequest.gdprConsent.gdprApplies));
+      }
+      if (typeof bidderRequest.gdprConsent.consentString !== 'undefined') {
+        deepSetValue(request, 'user.ext.consent', bidderRequest.gdprConsent.consentString);
+      }
+    }
+
+    if (typeof bidderRequest.tid !== 'undefined') {
```

Comment:
I am not aware of a `tid` field in bidder requests - the closest analog is `auctionId`

Reply:
Same as eids, it gets added to bidderRequest [here](https://github.com/prebid/Prebid.js/pull/11486/files/f5098b456ff0cd6bf94951d08fb91c61ae4a06db#diff-74bafa18f954c55e3036d5a4f29828357e9f3a536c0a62e7f51b1dedc4971648R402)

Reply:
You're right - I missed it.